### PR TITLE
feat(fluvio-cluster): prompt when deleting cluster

### PR DESCRIPTION
**Pull request by PanGan21**
Adds prompt when a user tries to delete a cluster.
Related: #3493

### CODE REVIEW

**Comment by digikata** [comment]
There is a little interaction on our existing CI tests that delete a cluster:

https://github.com/infinyon/fluvio/actions/runs/9271396243/job/25509183855?pr=4034#step:16:258

It may be as simple as adding `--force` to the test, or there might be something that will take a little more digging to understand.

File: crates/fluvio-cluster/src/cli/delete.rs [hunk 1]
```diff
@@ -19,10 +21,38 @@ pub struct DeleteOpt {
...
     sys_only: bool,
+
+    /// Do not prompt for confirmation
+    #[arg(long)]
+    force: bool,
 }
 
 impl DeleteOpt {
     pub async fn process(self) -> Result<()> {
+        if !self.force {
```
Comment (digikata) [inline crates/fluvio-cluster/src/cli/delete.rs:32]: Can this also output a message if `self.force == true` that `cluster/endpoint` is being deleted.

Comment (PanGan21) [reply crates/fluvio-cluster/src/cli/delete.rs]: Done!

**Comment by PanGan21** [comment]
> There is a little interaction on the existing CI tests that delete a cluster. It may be as simple as adding `--force` to the test, or there might be something that will take a little more digging to understand.
> 
> https://github.com/infinyon/fluvio/actions/runs/9271396243/job/25509183855?pr=4034#step:16:258
> 
> ```
> bats ./tests/cli/fluvio_smoke_tests/non-concurrent/cluster-delete.bats
> 1..1
> /usr/bin/timeout
> /usr/bin/timeout
> /usr/bin/timeout
> /usr/bin/timeout
> /usr/bin/timeout
> not ok 1 Delete the cluster
> # (from function `assert_success' in file tests/cli/fluvio_smoke_tests/non-concurrent/../../test_helper/bats-assert/src/assert_success.bash, line 42,
> #  in test file tests/cli/fluvio_smoke_tests/non-concurrent/cluster-delete.bats, line 30)
> #   `assert_success' failed
> #
> # -- command failed --
> # status : 1
> # output (6 lines):
> #   error: unexpected argument '--local' found
> #
> #   Usage: fluvio-dev cluster delete [OPTIONS]
> #
> #   For more information, try '--help'.
> #   IO error: not a terminal
> # --
> ```

I am investingating this.

**Comment by PanGan21** [comment]
Hi @digikata , it seems that `--local` local flag doesn't exist as an option. Should I remove it from the script as well?
```
Uninstall a Fluvio cluster

Usage: fluvio cluster delete [OPTIONS]

Options:
      --namespace <Kubernetes namespace>  
      --k8                                Remove only k8 fluvio installation
      --sys                               Remove system chart only
      --force                             Do not prompt for confirmation
  -h, --help                              Print help
```

**Comment by digikata** [comment]
Looking through the history it looks like at this level "--local" became the implicit default, with "--k8" optionally used if the backing cluster is hosted in k8s. But that's confusing because how was this test passing before.  I'm looking into that.

**Comment by digikata** [comment]
This PR is looking good for the core functaion, the "--local" error is part of the test that was bridge test logic between a version where there was a "--local" flag, and where it was removed.  For example, in `cluster-delete.bats` there is a test line:

```
    run bash -c "$FLUVIO_BIN cluster delete --local || $FLUVIO_BIN cluster delete"
```

The first delete fails, and the second is run, but a spurious error line is generated in that process.       

I did a cleanup pass on the tests https://github.com/infinyon/fluvio/tree/ac/feat-3493-prompt-cluster-delete,  I wonder if you could cherry pick that into your PR and we'll see if the tests are happy with those changes when we go to merge.

**Comment by PanGan21** [comment]
> This PR is looking good for the core functaion, the "--local" error is part of the test that was bridge test logic between a version where there was a "--local" flag, and where it was removed. For example, in `cluster-delete.bats` there is a test line:
> 
> ```
>     run bash -c "$FLUVIO_BIN cluster delete --local || $FLUVIO_BIN cluster delete"
> ```
> 
> The first delete fails, and the second is run, but a spurious error line is generated in that process.
> 
> I did a cleanup pass on the tests https://github.com/infinyon/fluvio/tree/ac/feat-3493-prompt-cluster-delete, I wonder if you could cherry pick that into your PR and we'll see if the tests are happy with those changes when we go to merge.

Thank you @digikata . I cherry picked your commit! Let's see the results of the ci!

**Comment by digikata** [comment]
Looks like `cluster-delete.bats` should revise the delete fallback logic to `fluvio cluster delete --force || fluvio cluster delete` ...  this is due to the cross version tests in the CI.. Do you want to make this change @PanGan21?

**Review by digikata — APPROVED** [review]
Looks like all the tests are passing now, I'll mark this to merge.

Thanks @PanGan21!

**Pull request merged**